src/build/emitter.ts

```diff
@@ -387,17 +388,51 @@ export function emitWebIdl(
             : obj.type;
         types.push(...(obj.additionalTypes ?? []).map((t) => ({ type: t })));
 
+        // propagate `any`
         const converted = types.map(convertDomTypeToTsTypeWorker);
-        const isAny = converted.some((t) => t === "any");
-        return isAny ? "any" : converted.join(" | ");
+        if (converted.includes("any")) return "any";
+
+        // convert `ArrayBuffer | SharedArrayBuffer` into `ArrayBufferLike` to be pre-ES2017 friendly.
+        const arrayBufferIndex = converted.indexOf("ArrayBuffer");
+        if (arrayBufferIndex >= 0) {
+          const sharedArrayBufferIndex = converted.indexOf("SharedArrayBuffer");
+          if (sharedArrayBufferIndex >= 0) {
+            const maxIndex = Math.max(arrayBufferIndex, sharedArrayBufferIndex);
+            const minIndex = Math.min(arrayBufferIndex, sharedArrayBufferIndex);
+            converted[minIndex] = "ArrayBufferLike"; // replace whichever comes first
+            converted.splice(maxIndex, 1); // drop whichever comes last
+          }
+        }
+        return converted.join(" | ");
       }
     }
 
     const type = convertBaseType();
-    const subtypeString = arrayify(obj.subtype)
+    let subtypeString = arrayify(obj.subtype)
       .map(convertDomTypeToTsType)
       .join(", ");
 
+    if (!subtypeString && compilerBehavior.genericTypedArrays) {
+      switch (type) {
+        case "ArrayBufferView":
+        case "DataView":
+        case "Int8Array":
+        case "Uint8Array":
+        case "Uint8ClampedArray":
+        case "Int16Array":
+        case "Uint16Array":
+        case "Int32Array":
+        case "Uint32Array":
+        case "Float32Array":
+        case "Float64Array":
+        case "BigInt64Array":
+        case "BigUint64Array":
+        case "Float16Array":
```

Comment:
Maybe make a subset of `bufferSourceTypes` in helpers? Otherwise it's prone to break when a new buffer source type appears.